rfcs/20191016-dlpack-support.md

```diff
@@ -0,0 +1,122 @@
+# dlpack support for interoperability with other GPU frameworks
+
+| Status        | (Proposed)       |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | 180 (https://github.com/tensorflow/community/pull/180) (update when you have community PR #)|
+| **Author(s)** | eoldridge@nvidia.com, wmjlyjemaine@gmail.com, zhoujinjing09@gmail.com |
+| **Sponsor**   | apassos@google.com, sanjoy@google.com                 |
+| **Updated**   | 2019-11-26                                           |
+
+## Objective
+
+This document proposes the adoption of dlpack (https://github.com/dmlc/dlpack) as way of passing tensor data to other frameworks without leaving the GPU and without a copy per [24453](https://github.com/tensorflow/tensorflow/issues/24453).  dlpack is a community effort to define a common tensor data structure that can be shared by different frameworks. dlpack is currently supported by cuPy, cuDF, DGL, TGL, PyTorch, and MxNet. 
+
+The interoperability of dlpack would allow for fast on-GPU communication between TensorFlow and these frameworks opening up a wide range of use cases outlined below.  It would further enable \_\_cuda_array_interface\_\_ interoperability through cuPy/cuDF which support both methods providing a way to transfer data to Numba, PyArrow and other frameworks that have adopted that method, although [a similar request has been made to support that method of interoperability](https://github.com/tensorflow/tensorflow/issues/29039) and ideally both would be supported.
+
+A solution has already been developed by @VoVAllen and @jermainewang (coauthored above) as an external python package.  This RFC would see the concepts from the package integrated into Tensorflow Core, and reviewed and enhanced by the TF team so that dlpack support is native.
+
+## Motivation
+
+DLPack is a community effort to define a common tensor data structure that can be shared by different frameworks allowing data to be quickly shared often with zero or minimal copy. One of the main bottlenecks when trying to achieve GPU performance when operating across different frameworks is I/O and data formatting.  The transfer of data between GPU and CPU or between formats is costly to the point where many operations become faster to simply run on the CPU because of the additional costs associated with moving/transforming the data.  Even when mechanisms exist to copy data without leaving the GPU, memory constraints limit the application because two copies of the data are required.  By implementing dlpack within TensorFlow there would be a way to transfer data directly between frameworks, enabling the development of a range of applications that weren't previously possible.
+
+Existing applications that take advantage of dlpack include: (adding my own and those listed in , other contributions needed)
```

Comment:
`(adding my own and those listed in , other contributions needed)` seems malformed